### PR TITLE
fix/authentication/signin-ui-tests : Fix and complete UI tests for SignInScreen

### PR DESCRIPTION
**Pull request by charliemangano**
# Fix and complete UI tests for SignInScreen
## Description
This PR introduces new tests for the SignIn screen and fixes the old ones. It also fixes the problems pointed out by said new tests. It closes issue #71 and fixes part of #63 .
## Changes
Added dependencies to make mockito work in androidTests (this was also done in `main` and later merged into this branch, so I'll ignore it in the rest of this PR).
Properly mocked `NavigationActions` and added verification of correct navigation. Also added checks for the correct calls to the `AuthViewModel` on valid attempts and for no calls to the `AuthViewModel` on invalid attempts. Removed irrelevant tests (tested for behavior of VM, not the point here). 
Invalid attempts are ones which have an empty email or password.
Fixed the problems pointed out by the added tests (two calls to `logInWithEmail` instead of one on valid attempt, call to `navigationAction` not detected when trying to sign up).
#### Added
None
#### Modified
- `app/src/androidTest/java/com/android/periodpals/ui/alert/SignInScreenTest.kt`: modified, removed and added tests
- `app/src/main/java/com/android/periodpals/ui/alert/SignInScreen.kt`: fixed problems pointed out by new tests
#### Removed
None
## Dependencies Added
None
## Testing
All UI tests for `SignInScreen` pass. They should be complete enough.

### CODE REVIEW

File: app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt [hunk 1]
```diff
@@ -1,144 +1,127 @@
...
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+
+class SignInScreenTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var authViewModel: AuthViewModel
```
Comment (coaguila) [inline app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt:29]: Perhaps a nitpick, you can add a `@Mock` tag to automatically innit mock classes without having to do it on `setUp()`. This may make the code cleaner.

File: app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt [hunk 2]
```diff
@@ -1,144 +1,127 @@
...
+
+    composeTestRule.onNodeWithTag("signInEmail").performTextInput(email)
+    composeTestRule.onNodeWithTag("signInPassword").performTextInput(password)
+    composeTestRule.onNodeWithTag("signInButton").performClick()
+    verify(authViewModel).logInWithEmail(any(), eq(email), eq(password))
+  }
+
+  @Test
+  fun signUpHereNavigatesToSignUpScreen() {
+    composeTestRule.onNodeWithTag("signInNotRegistered").performClick()
```
Comment (coaguila) [inline app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt:124]: Perhaps a nitpick but maybe add a space above to make it like the other tests.

**Review by coaguila — APPROVED** [review]
There's a couple of comments but are all nitpicks but overall LGTM.

**Review by taghizadlaura — CHANGES_REQUESTED** [review]
## Summary
Please see comments.

## Important

#### Code Quality
- LGTM

#### Functionality
- See comments

## Testing
- See comments

## Documentation
- LGTM

## Suggestions
- Maybe add more tests to cover edge cases, such as network errors during login.

## Steps before PR approved
- See comments

**Comment by charliemangano** [comment]
> Maybe add more tests to cover edge cases, such as network errors during login.

This should be handled at the VM level, so those tests would be out of the scope of this PR.

**Review by taghizadlaura — APPROVED** [review]
LGTM now
